libraries/chain/include/steem/chain/database.hpp

```diff
@@ -489,8 +489,11 @@ namespace steem { namespace chain {
          /**
           * @return a list of available NAIs.
          */
-         vector< asset_symbol_type > get_smt_next_identifier();
+         vector< asset_symbol_type > get_nai_pool();
```

Comment:
As best I can tell, this is only needed for the API and testing. Let's not have it as a method on the database then. Also, if the NAI pool object is a singleton, this vector is stored on the object and easily retrievable by any code that needs it without extra work.